[MSPFormat] Can't read Problem Instance 10 (Inventory Problem with Lead Time) from MSPLib-Library

Using the collection of problem in the [MSPLib-Library](https://github.com/bonnkleiford/MSPLib-Library), I ran into some problems reading the problem instance 10, the Inventory Problem with lead time, using 

`model = SDDP.MSPFormat.read_from_file(path)`

with the appropriate path, which lead to
`ERROR: AssertionError: term["stage"] == stage - 1`

It occurs on all problem instances for problem 10, for every discretization.
Other problems, such as 4, 9 or 18 work without a problem.

I used SDDP.jl v1.7.0

--- a/src/MSPFormat.jl
+++ b/src/MSPFormat.jl
@@ -177,9 +177,14 @@ function _build_lhs(stage::Integer, sp::JuMP.Model, terms::Vector{Any})
             @assert x isa SDDP.State
             if term["stage"] == stage
                 x = x.out
-            else
-                @assert term["stage"] == stage - 1
+            elseif term["stage"] == stage - 1
                 x = x.in
+            else
+                error(
+                    "SDDP.jl does not support this MSPFormat file because it " *
+                    "contains state variables from stages other than `t` or " *
+                    "`t-1`. Got `t-$(stage - term["stage"])`"
+                )
             end
         end
         coef = _get_constant(term["coefficient"])

--- a/src/MSPFormat.jl
+++ b/src/MSPFormat.jl
@@ -183,7 +183,7 @@ function _build_lhs(stage::Integer, sp::JuMP.Model, terms::Vector{Any})
                 error(
                     "SDDP.jl does not support this MSPFormat file because it " *
                     "contains state variables from stages other than `t` or " *
-                    "`t-1`. Got `t-$(stage - term["stage"])`"
+                    "`t-1`. Got `t-$(stage - term["stage"])`",
                 )
             end
         end

--- a/test/MSPFormat.jl
+++ b/test/MSPFormat.jl
@@ -133,6 +133,22 @@ function test_electric()
     return
 end
 
+function test_stage_lead_time()
+    sp = JuMP.Model()
+    sp[:x] = SDDP.State(@variable(sp), @variable(sp))
+    terms = Any[
+        Dict{String, Any}("name"=>"x", "stage"=>2, "coefficient"=>Any[1.0])
+        Dict{String, Any}("name"=>"x", "stage"=>0, "coefficient"=>Any[-1.0])
+    ]
+    @test_throws(
+        ErrorException(
+            "SDDP.jl does not support this MSPFormat file because it contains state variables from stages other than `t` or `t-1`. Got `t-2`",
+        ),
+        MSPFormat._build_lhs(2, sp, terms),
+    )
+    return
+end
+
 end  # module
 
 TestMSPFormat.runtests()

--- a/test/MSPFormat.jl
+++ b/test/MSPFormat.jl
@@ -134,11 +134,11 @@ function test_electric()
 end
 
 function test_stage_lead_time()
-    sp = JuMP.Model()
+    sp = Model()
     sp[:x] = SDDP.State(@variable(sp), @variable(sp))
     terms = Any[
-        Dict{String, Any}("name"=>"x", "stage"=>2, "coefficient"=>Any[1.0])
-        Dict{String, Any}("name"=>"x", "stage"=>0, "coefficient"=>Any[-1.0])
+        Dict("name" => "x", "stage" => 2, "coefficient" => [1.0]),
+        Dict("name" => "x", "stage" => 0, "coefficient" => [-1.0]),
     ]
     @test_throws(
         ErrorException(

